droidlet/tools/hitl/nsp_retrain/scripts/sweep_runner.py

```diff
@@ -0,0 +1,219 @@
+        all_arglists.append(a + static_args)
+
+    # TODO call the data script automatically here
+    model_out = os.path.join(output_dir, "model_out/")
+    os.makedirs(model_out, exist_ok=True)
+
+    print("the sweep will use data in " + target_data_folder)
+    print("models will be written to " + model_out)
+
+    train_command = (
+        "droidlet/perception/semantic_parsing/nsp_transformer_model/train_model.py"
+        + " --data_dir "
+        + target_data_folder
+        + " --output_dir "
+        + model_out
+        + " "
+    )
+
+    errpaths = []
+    outpaths = []
+    modelpaths = []
+    for i in range(len(all_arglists)):
+        uid = job_name + "_P" + str(i)
+        body = "#! /bin/bash \n"
+        body += "#SBATCH --job-name=" + uid + "\n"
+        body += "#SBATCH --output=" + os.path.join(output_dir, str(i) + ".out") + "\n"
+        outpaths.append(os.path.join(output_dir, str(i) + ".out"))
+        body += "#SBATCH --error=" + os.path.join(output_dir, str(i) + ".err") + "\n"
+        errpaths.append(os.path.join(output_dir, str(i) + ".err"))
+        body += "#SBATCH --partition=" + opts.partition + "\n"
+        body += "#SBATCH --nodes=1 \n"
+        body += "#SBATCH --ntasks-per-node=1 \n"
+        body += "#SBATCH --gres=gpu:1 \n"
+        body += "#SBATCH --cpus-per-task=10 \n"
+        body += "#SBATCH --signal=B:USR1@60 #Signal is sent to batch script itself \n"
+        body += "#SBATCH --open-mode=append \n"
+        body += "#SBATCH --time=4320 \n"
+        body += "\n"
+        #    body += st #FIXME
+        body += "\n"
+        body += "module purge\n"
+        # TODO make a env loader...
+        body += "module load NCCL/2.2.13-cuda.9.0 \n"
+        body += "module load anaconda3/5.0.1 \n"
+        body += "source activate /private/home/ethancarlson/.conda/envs/droidlet_env \n"
+        #######
+        body += "cd " + opts.droidlet_dir + " \n"
+        body += (
+            "/private/home/ethancarlson/.conda/envs/droidlet_env/bin/ipython -- " + train_command
+        )
+        body += all_arglists[i]
+        # body += " --sbatch --save_model_uid " + job_name + "_" + str(i)
+        scriptname = os.path.join(scripts_dir, str(i) + ".sh")
+        g = open(scriptname, "w")
+        g.write(body)
+        g.close()
+        st = os.stat(scriptname)
+        os.chmod(scriptname, st.st_mode | stat.S_IEXEC)
+
+    mastername = os.path.join(scripts_dir, "master")
+    g = open(mastername, "w")
+    g.write("#! /bin/sh \n")
+    for i in range(len(all_arglists)):
+        g.write("# opts :: " + varying_args[i] + " :: " + outpaths[i] + "\n")
+    for i in range(len(all_arglists)):
+        g.write("echo " + "'" + varying_args[i] + " :: " + outpaths[i] + "'" + "\n")
+    for i in range(len(all_arglists)):
+        g.write("sbatch " + str(i) + ".sh &\n")
+    g.close()
+    st = os.stat(mastername)
+    os.chdir(scripts_dir)
+    os.chmod(mastername, st.st_mode | stat.S_IEXEC)
+    os.system(mastername)
```

Comment:
Thanks for adding this file in. Can you adda README to the `nsp_retrain` folder highlighting the purpose of the new files ?